.github/workflows/test.yml

```
@@ -1,13 +1,23 @@
 name: "Test"
 on:
   pull_request:
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
 jobs:
   nix-build:
     name: 'Nix / Build'
     strategy:
       matrix:
-        os: [ubuntu-latest, macos-latest]
-    runs-on: ${{ matrix.os }}
+        include:
```

Note fail fast is enabled here. If you wish to let that happen this looks good.